### PR TITLE
change: [M3-8209] - Update Region label globally

### PR DESCRIPTION
**Pull request by hana-linode**
## Description 📝
Follow-up to https://github.com/linode/manager/pull/10702 except without the feature flagging due to additional complexity for testing. This PR also includes state in the label for regions in the US

## Changes  🔄
List any change relevant to the reviewer.
- Display Country, State, City for regions in the US
- Display Country, City for all other regions

## Target release date 🗓️
8/19. This change will be rolled out globally and separate from Gecko (approved by Lew)

## Preview 📷
| Before  | After   |
| ------- | ------- |
| image | image |
| image | image |

### Verification steps
(How to verify changes)
- Add  `gecko2: {
      enabled: false,
    },` to `useFlags.ts` to turn Gecko features off locally
  - We do not want to change the LD flag since other teams rely on it
- Check region labels throughout the app. The label should display as Country, State, City for regions in the US and Country, City for other regions.

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Comment by jdamore-linode** [comment]
@hana-linode I'll take a second pass at those tests later this week!

**Comment by jdamore-linode** [comment]
Hey @hana-linode, just pushed a couple Cypress changes that I think should resolve a lot of the test failures. I expect there'll still be some issues, so I'll circle back later today and tackle any remaining failures once CI finishes up

**Review by mjac0bs — APPROVED** [review]
Spot checked Region throughout the CM products to confirm the new format and our test coverage is pretty diligent in checking for this as well, which is all passing now. (Appreciate the thorough documentation of new utils too, thanks @jdamore-linode  and @hana-linode for fixing.)

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/HostNamesList.test.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ describe('HostNamesList', () => {
     renderWithTheme(<HostNamesList objectStorageKey={mockObjectStorageKey} />);
 
     const copyButton = await screen.findByLabelText(
-      'Copy Fake Region, NC: http://example.com to clipboard'
+      'Copy US, Fake Region, NC: http://example.com to clipboard'
```
Comment (coliu-akamai) [inline packages/manager/src/features/ObjectStorage/AccessKeyLanding/HostNamesList.test.tsx:36]: I love this region example 😆

**Review by coliu-akamai — APPROVED** [review]
✅ confirmed region labels updated for places I checked throughout the app
++ on documentation! 🎉

**Pull request merged**